activerecord/lib/active_record/asynchronous_queries_tracker.rb

```diff
@@ -33,7 +41,7 @@ def complete(asynchronous_queries_tracker)
     attr_reader :current_session
 
     def initialize
-      @current_session = nil
+      @current_session = NoSession
```

Comment:
After some thought, I think I make sense to have `defer` work by default. It's good to cancel queries when we know we exited the context in which they are useful, but if we're in `rails console` or just a random `rails runner` script, there's no reason to not allow it.